libs/community/langchain_community/embeddings/self_hosted_hugging_face.py

```diff
@@ -34,6 +34,15 @@ def load_embedding_model(model_id: str, instruct: bool = False, device: int = 0)
 
         client = INSTRUCTOR(model_id)
 
+    if importlib.util.find_spec("habana_frameworks") is not None:
+        import habana_frameworks.torch.hpu as hthpu
+
+        if hthpu.is_available():
+            import torch
+            from habana_frameworks.torch.hpu import wrap_in_hpu_graph
+            client = wrap_in_hpu_graph(client)
+            return client.eval().to(torch.device(device))
```

Comment:
Does this work as expected? I see that device here is an integer instead of "hpu".

Reply:
@sjagtap1803 indeed it it better to run it with:  
```python
return client.eval().to(torch.device("hpu"))
```
Changed it.

Reply:
Thanks